edb/edgeql/compiler/viewgen.py

```diff
@@ -899,6 +899,18 @@ def _raise_on_missing(
         )
 
 
+@dataclasses.dataclass(kw_only=True, repr=False, eq=False)
+class RewriteContext:
+    specified_ptrs: Set[sn.UnqualName]
+    kind: qltypes.RewriteKind
+
+    base_type: s_objtypes.ObjectType
+    shape_type: s_objtypes.ObjectType
+
+    # Cache for computing `prepare_rewrite_anchors`, since that is expensive
+    anchor_cache: Dict[s_objtypes.ObjectType, RewriteAnchors]
```

Comment:
This seems unused?

Reply:
Oh, yeah, I wanted to used this instead of a local anchors variable, but forgot about it.

It is not really any better, so your commit is fine.